x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysColumns.java

```diff
@@ -169,7 +173,7 @@ private static void fillInRows(String clusterName, String indexName, Map<String,
             DataType type = field.getDataType();
 
             // skip the nested, object and unsupported types
-            if (isPrimitive(type)) {
+            if (isPrimitive(type) && isTypeSupportedInVersion(type, version)) {
```

Comment:
I believe this is incorrect - any column using of an unsupported type should be exposed as such - unsupported, not hidden from view.

Reply:
@costin `isPrimitive` is already eliminating the `UNSUPPORTED` types from `sys columns`. If `unsigned_long` would have been configured as `UNSUPPORTED` instead, it wouldn't have been part of `sys columns` output by default.
`SHOW COLUMNS` on the other hand should display `unsupported` for an old driver version and `unsigned_long` for a newer driver version. Which, I don't think it happens atm.

Reply:
Right. And we do that to hide these fields from "nosy" apps, that would otherwise still SELECT them (since `UNSUPPORTED` is just a vendor-specific type, like the rest of the types).